### PR TITLE
Add Windows Forms TaskDialog API sample

### PR DESCRIPTION
**Pull request by RussKie**
## Summary

Add a sample for Windows Forms TaskDialog API, that was added in .NET 5.0 in https://github.com/dotnet/winforms/pull/1133.

Credits: this sample was authored by @kpreisser in https://github.com/kpreisser/TaskDialogDemo, the author of [`Task Dialog` for Windows Forms](https://github.com/dotnet/winforms/pull/1133).

### CODE REVIEW

**Comment by adegeo** [comment]
Thanks for adding this here! This sample should be split into two, a csharp one and a vb one. Basically, add a readme in the root directory for each sample, it should contain a metadata header so that the sample shows up in the samples browser. Here is an example of the header: https://raw.githubusercontent.com/dotnet/samples/master/mef/simple-calculator/vb/readme.md

Which looks like this:
```
---
languages:
- vb
products:
- dotnet-core
- windows
page_type: sample
name: "MEF Sample: Simple Calculator (Visual Basic)"
urlFragment: "simple-calculator-vb"
description: "A .NET Core console application that contains the example method from Simple Calculator MEF Application in Visual Basic."
---
```

**Comment by RussKie** [comment]
> This sample should be split into two, a csharp one and a vb one.

Sure, but it would make it more difficult to keep the sample in sync between the languages.

**Comment by RussKie** [comment]
Updated, please have a look

**Comment by adegeo** [comment]
Thanks! Which preview of .NET 5 is required?

**Comment by RussKie** [comment]
.NET 5.0 Preview 4

File: windowsforms/TaskDialogDemo/cs/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+This sample is written in C# and targets .NET 5.0 running on Windows. It requires the [.NET 5.0 SDK](https://dotnet.microsoft.com/download/dotnet/5.0).
+
+## Building the sample
+
+Build and run the demo:
+
+```
+TaskDialogDemo\cs> dotnet run
+```
```
Comment (adegeo) [inline windowsforms/TaskDialogDemo/cs/README.md:31]: Changing "Building" section to be similar to what is used by the other examples.

```suggestion
## Building the sample

To download and run the sample, follow these steps:

01. Download and unzip the sample.

01. In Visual Studio 2019 or later:

    01. On the menu bar, choose **File** > **Open** > **Project/Solution**.

    01. Navigate to the folder that holds the unzipped sample code, and open the project (*.csproj*) file.

    01. Choose the F5 key, or Ctrl+F5 keys, to run the project.

01. From the command line:

    01. Navigate to the folder that holds the unzipped sample code.

    01. At the command line, type `dotnet run`.
    
```

File: windowsforms/TaskDialogDemo/vb/TaskDialogDemo.sln [hunk 2]
```diff
@@ -0,0 +1,25 @@
+﻿
```
Comment (adegeo) [inline windowsforms/TaskDialogDemo/vb/TaskDialogDemo.sln:1]: We don't include SLN files unless they are absolutely required. We can safely delete this file.

File: windowsforms/TaskDialogDemo/vb/README.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+This sample is written in Visual Basic and targets .NET 5.0 running on Windows. It requires the [.NET 5.0 SDK](https://dotnet.microsoft.com/download/dotnet/5.0).
+
+## Building the sample
+
+Build and run the demo:
+
+```
+TaskDialogDemo\vb> dotnet run
+```
```
Comment (adegeo) [inline windowsforms/TaskDialogDemo/vb/README.md:31]: Changing "Building" section to be similar to what is used by the other examples.

```suggestion
## Building the sample

To download and run the sample, follow these steps:

01. Download and unzip the sample.

01. In Visual Studio 2019 or later:

    01. On the menu bar, choose **File** > **Open** > **Project/Solution**.

    01. Navigate to the folder that holds the unzipped sample code, and open the project (*.vbproj*) file.

    01. Choose the F5 key, or Ctrl+F5 keys, to run the project.

01. From the command line:

    01. Navigate to the folder that holds the unzipped sample code.

    01. At the command line, type `dotnet run`.
    
```

File: windowsforms/TaskDialogDemo/cs/TaskDialogDemo.sln [hunk 4]
```diff
@@ -0,0 +1,25 @@
+﻿
```
Comment (adegeo) [inline windowsforms/TaskDialogDemo/cs/TaskDialogDemo.sln:1]: Forgot this one too, We can delete this file.

**Comment by RussKie** [comment]
done

**Comment by adegeo** [comment]
I'm going to hold off on merging this. I have a PR that enables our CI testing and I need to update it for .NET 5, so this PR is a great test case. Is that OK? Let me know. Thanks!

**Comment by RussKie** [comment]
No worries. No rush on my part. Thank you

**Pull request merged**